### DISCUSSION TITLE
Best approach to handle every change in a data model class

### DISCUSSION THREAD

**Discussion by maslewandowski** [CommunityToolkit/Maui#1943] [Q&A]
Hi,
I'm trying to save a data model object from a data model class to a json file.

**This is the process:**
1. The user fills out the data model object by interacting with a form in a .net maui application on ios. The binding is provided by communitytoolkit.mvvm
2. If there is any change in ANY property in the main data model object, then I want to save the data model object to a json file serialized with newtonsoft.

quick nonsense example:
**Main Data Model:**
```
public partial class MainDataModel : ObservableObject
{
    [ObservableProperty]
    private string mainStringProperty;
    [ObservableProperty]
    private bool mainBoolProperty;

    [ObservableProperty]
    private SubDataModel subDataModel;

    partial void OnMainBoolPropertyChanged(bool oldValue, bool newValue)
    {
        throw new NotImplementedException();
    }
    partial void OnMainStringPropertyChanged(string? oldValue, string newValue)
    {
        throw new NotImplementedException();
    }
}
```

**Sub Data Model**
```
public partial class SubDataModel : ObservableObject
{
    [ObservableProperty]
    private string stringProperty;
    [ObservableProperty]
    private bool boolProperty;

    partial void OnBoolPropertyChanged(bool oldValue, bool newValue)
    {
        throw new NotImplementedException();
    }

    partial void OnStringPropertyChanged(string? oldValue, string newValue)
    {
        throw new NotImplementedException();
    }
}
```

What is the best approach, if i want to call a method when **ANYTHING** changes in this example data model class structure?
Would you recommend another approach to reach my goal as described in process?

Thanks!

**Comment by pictos** [comment]
Doing that on each propertyChanged will be really expensive, also creating json + saving it on db can be done on a backgroundThread, that way your UI will not be blocked. I recommend using the [StoppedBehavior](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/behaviors/user-stopped-typing-behavior), that way you can set a delay between the fire and the execution of your code

**Reply by bijington** [reply]
I agree with @pictos that it could expensive to do it on property changed - this will be updated for every key press of a keyboard potentially.

You could use the StoppedBehavior or some level of debouncing approach to serialise the model out once the model has stopped being updated.
